pkg/aggregator/aggregator.go

```diff
@@ -849,6 +855,14 @@ func (agg *BufferedAggregator) run() {
 				}
 			}
 			tlmFlush.Add(1, event.eventType, state)
+		case event := <-agg.contlcycleIn:
+			aggregatorContainerLifecycleEvents.Add(1)
+			go func(contlcycle senderContainerLifecycleEvent) {
```

Comment:
should we use a channel instead and define only one goroutine that dequeue the message event to push it in the specific forwarder?

Reply:
I'm worried about the number of go routine we can generate if we have lots of events

Reply:
@remeh I'd like to hear from you on this: The current implem is inspired by what it's done for `orchestratorMetadata` (`use a routine to avoid blocking the aggregator` right above in the code). 
I know that there are plans to refactor/get rid of this big `select` in the aggregator, so my question is what is exactly the plan? what's your take on Cedric's suggestion? ideally we should have something working for the different pipelines.

Reply:
Considering current the `BufferedAggregator`:
  * I don't know how much events are planned to be processed here, but you have to think about the garbage the routines are generating. Spawning a lot of routines isn't free for the GC.
  * If it's only a few, I guess there is no real reasons to use a separate routine every time then, since this `select` won't select that case that much (i.e. not slowing down the rest)
  * Recently, the database-monitoring team had close to the same needs (new events + forwarder), you can actually see it right above your `case` and it's not leading to issues afaik. However, [they are indeed using a channel under-the-hood ](https://github.com/DataDog/datadog-agent/blob/main/pkg/epforwarder/epforwarder.go#L97)(reducing latency and to avoid blocking this select for too long) and you may want to do the same.

Considering the incoming `Demultiplexer` implementation:
  * In order to accommodate your use-case (and e.g. the database-monitoring one), I plan to have the Demultiplexer spawning a new routine just to have a `select` watching and processing these (avoiding "slowing down" the main sampling pipelines)

For your PR:
  * I would also be worried about how many routines you plan to have created per second here, I would indeed consider using a channel if latency matters and if the rest of the processing down the line can take some time (and slowing down this `select`),
  * It is the proper place to add this for now (the `Demultiplexer` isn't really opened to "extras" pipelines).

Let me know if you want me to elaborate more.

Reply:
Thanks for the input, these are valid points! There is actually no heavy processing in `SendContainerLifecycleEvent` but I'll add a channel just in case.

Reply:
addressed in the latest commit